vendor/brave-ios/Shared/BATCommonOperations.mm

```diff
@@ -102,16 +102,16 @@ - (void)loadURLRequest:(const std::string &)url headers:(const std::vector<std::
     if (error) {
       errorDescription = error.localizedDescription.UTF8String;
     }
-    // For some reason I couldn't just do `std::map<std::string, std::string> responseHeaders;` due to std::map's
-    // non-const key insertion
-    auto* responseHeaders = new std::map<std::string, std::string>();
+    // For some reason I couldn't just do `base::flat_map<std::string, std::string>
+    // responseHeaders;` due to base::flat_map's non-const key insertion
+    auto* responseHeaders = new base::flat_map<std::string, std::string>();
```

Comment:
Does this compile? I assume based on my original comment, I tried to use `base::flat_map` but couldn't

Reply:
I don't know yet, to be honest. I don't have a Mac around so, I focused mainly on Linux and Android so far and so I made this change in the .mm files as carefully as I could by reading the code and changing what I think needed changing (i.e. the `loadURLRequest()` method from `BATCommonOperations.{h,mm}`), and I was now hoping for the bots to tell me whether it builds or not.

In the other hand, I don't understand well what you mean by "I assume based on my original comment", I don't see a reference to `base::flat_map` at all there, just to `std::map`. Could you please clarify? Thanks!

Reply:
Your right, sorry, the find + replace for all the std::map's in that comment got me confused 😅